### PR TITLE
Added Agent methods and tests for: Get, List, Update and Delete

### LINKED ISSUE #24: Implement Agents
https://developers.freshdesk.com/api/#agents

### PR DESCRIPTION
**Pull request by davinthesmith**
fix #24

### CODE REVIEW

**Comment by davinthesmith** [comment]
I was unable to get the tests to run on my machine.  However, I used some existing tests to mock up a good start.

**Comment by davinthesmith** [comment]
Failure is occurring during npm install?  Not sure what the issue is.  I didn't touch anything that would affect npm install.

File: lib/client.js [hunk 1]
```diff
@@ -147,6 +147,24 @@ Freshdesk.prototype = Object.create({
 		makeRequest('GET', this._auth, `${this.baseUrl}/api/v2/contact_fields`, null, null, cb)
 	},
 
+	//Agents
+
```
Comment (maxkoryukov) [inline lib/client.js]: Could you add a JSDoc here ([example](https://github.com/arjunkomath/node-freshdesk-api/blob/master/lib/client.js#L46))? For all four new methods (actually, there should be 5 new methods).

Don't forget to add **filter** example for `listAllAgents`

File: test/api.agent.test.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
+/*
+Node wrapper for Freshdesk v2 API
+
+Copyright (C) 2016-2017 Maksim Koryukov <maxkoryukov@gmail.com>
```
Comment (maxkoryukov) [inline test/api.agent.test.js]: Add your name in the next line;)

File: test/api.agent.test.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+
+You should have received a copy of the MIT License along with this
+program. If not, see <https://opensource.org/licenses/MIT>.
+
+http://spdx.org/licenses/MIT
+*/
+
+"use strict";
```
Comment (maxkoryukov) [inline test/api.agent.test.js]: `'use strict'`

Single quotes and no semicolon (I will fix other occurrences)

File: lib/client.js [hunk 4]
```diff
@@ -147,6 +147,24 @@ Freshdesk.prototype = Object.create({
...
+	},
+
+	updateAgent(id, data, cb) {
+		makeRequest('PUT', this._auth, `${this.baseUrl}/api/v2/agents/${id}`, null, data, cb)
+	},
+
+	deleteAgent(id, cb) {
+		makeRequest('DELETE', this._auth, `${this.baseUrl}/api/v2/agents/${id}`, null, null, cb)
+	},
+
```
Comment (maxkoryukov) [inline lib/client.js]: Seems like you forgot to implement the remaining one method [currentAgent](https://developers.freshdesk.com/api/#me)

**Comment by maxkoryukov** [comment]
Hello, @davinthesmith !
Thanks for your interest in this project, and for your contribution :wink:

Another one notices about this PR:
* add yourself to the list of authors at the end of `readme.md`

----

> Failure is occurring during npm install? Not sure what the issue is. I didn't touch anything that would affect npm install.

Don't worry, I will check this failure

**Comment by maxkoryukov** [comment]
@davinthesmith build on travis fixed, detail are here: #29

**Comment by maxkoryukov** [comment]
@davinthesmith still opened:

- [ ] js doc for new methods
- [ ] authors in readme is not updated
- [ ] [currentAgent](https://developers.freshdesk.com/api/#me) still not implemented.

And you have `merged` changes, instead of `rebase`. This step had converted this job (merging PR) to hell. 
Plz, read this: https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request
It is a very good article about `rebase` and git in general. Hope, it could be useful for you;)

**Comment by Velua** [comment]
I've just made some updates and created a pull request here...
https://github.com/davinthesmith/node-freshdesk-api/pull/2

This is my first proper pull request so I might have done it wrong....

**Comment by Velua** [comment]
I've gone and done both now #32 
Let me know which one was better practice!

**Comment by davinthesmith** [comment]
@maxkoryukov , sorry about the merge.  I wasn't sure which bit to rebase and which bit to merge.  I rebased my master branch and merged the changes into it, but apparently that was incorrect.  I was using SourceTree for this, so I'm not sure if something there went sideways.

**Comment by maxkoryukov** [comment]
@davinthesmith don't worry;) I've sent you a link just because time ago that guide had helped me to understand how `git-rebase` works. Anyway, now that `merge` is not a problem.

@Velua - I have reviewed that PR too;)

**Comment by maxkoryukov** [comment]
Move all further conversations to #32

**Comment by maxkoryukov** [comment]
All this changes were merged in #32

**Pull request closed without merging**